### PR TITLE
Move initial Node.js polyfills to workerd

### PR DESCRIPTION
**Pull request by jasnell**
@mikea the key pieces to review here are the changes to the bazel build file and the tsconfig. Everything other than the sample modifications are a straight copy of the TS files.

### CODE REVIEW

File: src/node/BUILD.bazel [hunk 1]
```diff
@@ -30,6 +30,6 @@ wd_api_bundle(
     internal_modules = dict([(
         m.removesuffix(".ts") + ".js",
         "node-internal:" + m.removeprefix("internal/").removesuffix(".ts"),
-    ) for m in internal_modules]),
+    ) for m in internal_modules if not m.endswith(".d.ts")]),
```
Comment (mikea) [inline src/node/BUILD.bazel:33]: thanks for taking care of this.

File: src/node/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "target": "ESNext",
     "module": "ESNext",
-    "lib": [ "ESNext" ],
+    "lib": [ "ESNext", "DOM" ],
```
Comment (mikea) [inline src/node/tsconfig.json:5]: todo for myself: replace this with either smaller definition or use generated .d.ts types (preferred).

File: samples/nodejs-compat/worker.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { EventEmitter } from 'node:events';
```
Comment (mikea) [inline samples/nodejs-compat/worker.js:1]: you can also easily add an import test to https://github.com/cloudflare/workerd/blob/main/src/workerd/api/node/node-test.c%2B%2B (maybe for async context?)

**Review by mikea — APPROVED** [review]

File: samples/nodejs-compat/worker.js (hunk 3 above)
Comment (jasnell) [reply samples/nodejs-compat/worker.js:1]: Yeah, I'll be porting the tests we have in the internal ew-test files out to here separately

**Pull request merged**